test/docs/OpenJ9TestUserGuide.md

```diff
@@ -0,0 +1,302 @@
+<!--
+Copyright (c) 2016, 2018 IBM Corp. and others
+
+This program and the accompanying materials are made available under
+the terms of the Eclipse Public License 2.0 which accompanies this
+distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+or the Apache License, Version 2.0 which accompanies this distribution and
+is available at https://www.apache.org/licenses/LICENSE-2.0.
+
+This Source Code may also be made available under the following
+Secondary Licenses when the conditions for such availability set
+forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+General Public License, version 2 with the GNU Classpath
+Exception [1] and GNU General Public License, version 2 with the
+OpenJDK Assembly Exception [2].
+
+[1] https://www.gnu.org/software/classpath/license.html
+[2] http://openjdk.java.net/legal/assembly-exception.html
+
+SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+-->
+
+# How-to Run Tests
+
+Let's create the following scenario:
+For the purpose of this example, we assume you have an OpenJ9 SDK ready
+for testing. Below are the specific commands you'd run to compile and
+run tests. Details are explained in *Tasks in OpenJ9 Test* section below.
+
+```
+cd openj9/test/TestConfig
+export JAVA_BIN=<path to JDK bin directory that you wish to test>
+export SPEC=linux_x86-64_cmprssptrs
+make -f run_configure.mk
+make test
+```
+
+## Prerequisites:
+Please read [Prerequisites.md](./Prerequisites.md) for details.
+
+# Tasks in OpenJ9 Test
+
+1. Configure environment:
+
+  * required environment variables
+    ```
+    JAVA_BIN=<path to JDK bin directory that you wish to test>
+    SPEC=[linux_x86-64|linux_x86-64_cmprssptrs|...] platform_on_which_to_test
+    JAVA_VERSION=[SE80|SE90|SE100|Panama|Valhalla] (SE90 default value)
+    JAVA_IMPL=[openj9|hotspot|sap] (openj9 default value)
+    ```
+
+  * dependent libs for tests
+
+Please read [DependentLibs.md](./DependentLibs.md) for details.
+
+2. Compile tests:
+
+  * compile and run all tests
+    ```
+    make test
+    ```
+
+  * only compile but do not run tests
+    ```
+    export BUILD_LIST=comma_separated_projects_to_be_compiled (defaults to all projects)
+    make compile
+    ```
+
+3. Add more tests:
+
+  * for Java8/Java9 functionality
+
+    - If you have added new features to OpenJ9, you will likely
+    need to add new tests. Check out openj9/test/TestExample for
+    the format to use.
+
+    - If you have many new test cases to add, then you may want to
```

Comment:
If you have many new test cases to add and special build requirements, then you may want...